luaui/Widgets/Include/select_api.lua

```diff
@@ -0,0 +1,625 @@
+				local minHealth = minHealthPercent * udef.health
+				local health = spGetUnitHealth(uid)
+				return health > minHealth
+			end, minHealthPercent)
+			-- elseif token == "RulesParamEquals" then
+			-- 	local param = getNextToken()
+			-- 	local value = getNextToken()
+
+			-- 	if not value or not param then
+			-- 		break
+			-- 	end
+
+			-- 	local filterName = param .. "Rule"
+			-- 	filters[filterName] = invertCurry(invert, function(udef, _, uid, args)
+			-- 		local param = args.param
+			-- 		local value = args.value
+			-- 		-- implementation here?
+			-- 	end, {param = param, value = value})
+		elseif token == "AntiAir" then
+			filters.antiAir = invertCurry(invert, function(udef)
+				if udef.wDefs == nil or udef.canFly then
+					return false
+				end
+
+				for _name, weapondef in pairs(udef.wDefs) do
+					if (weapondef.damages[vtoldamagetag] > weapondef.damages[defaultdamagetag]) then
+						return true
+					end
+				end
+				return false
+			end)
+		elseif token == "WeaponRange" then
+			local minRange = tonumber(getNextToken())
+			if not minRange then
+				break
+			end
+
+			filters.weaponRange = invertCurry(invert, function(udef, _, _, minRange)
+				if udef.wDefs == nil then
+					return false
+				end
+
+				for _name, weapondef in pairs(udef.wDefs) do
+					if weapondef.range > minRange then
+						return true
+					end
+				end
+				return false
+			end, minRange)
+
+			-- string comparision
+		elseif token == "Category" then
+			local category = getNextToken()
+			if not category then
+				break
+			end
+
+			category = string.lower(category)
+
+			filters.category = invertCurry(invert, function(udef, _, _, category)
+				return udef.modCategories[category]
+			end, category)
+		elseif token == "IdMatches" then
+			local name = getNextToken()
+			if not name then
+				break
+			end
+
+			local udefid = nameLookup[name];
+
+			if udefid then
+				idMatchesSet[udefid] = true
+
+				-- requires special invert logic
+				-- treats `invert = false` as priority
+				-- we don't want to handle pointless edge cases like IdMatches_armcom_Not_IdMatches_armcom
+				-- on the other hand IdMatches_armcom_Not_IdMatches_armflea is basically the same as IdMatches_armcom
+				local skip = false
+				if invertIdMatches == nil or invertIdMatches == invert then
+					invertIdMatches = invert
+				elseif invertIdMatches == true then
+					idMatchesSet = {}
+					invertIdMatches = false
+				elseif invertIdMatches == false then
+					skip = true
+				end
+
+				if not skip then
+					filters.idMatches = invertCurry(invertIdMatches, function(_, udefid, _, idMatchesSet)
+						return idMatchesSet[udefid] or false
+					end, idMatchesSet)
+				end
+			end
+		elseif token == "NameContain" then
+			local name = getNextToken()
+			if not name then
+				break
+			end
+
+			filters.name = invertCurry(invert, function(udef, _, _, name)
+				return stringContains(udef.translatedHumanName, name)
```

Comment:
I had it that way originally, and agree that it isn't great. This is for backwards compatibility, in case there are people using this filter in their config.

Reply:
This is a buggy, fragile filter to even have, and we should remove it